deepchem/trans/transformers.py

```diff
@@ -972,6 +972,8 @@ def transform_array(self, X, y, w, ids):
       Transformed array of ids
     """
     w_balanced = np.zeros_like(w)
+    if len(y.shape) == 1 and len(w.shape) == 2 and w.shape[1] == 1:
```

Comment:
It might be good to add a unit test to document the fix this is making. Shape bugs are nasty so getting a nice set of unit tests to catch corner cases could help a lot

Reply:
Added!